lib/srv/db/proxyserver.go

```diff
@@ -270,6 +280,50 @@ func (s *ProxyServer) Proxy(ctx context.Context, clientConn, serviceConn io.Read
 	return trace.NewAggregate(errs...)
 }
 
+func (s *ProxyServer) monitorConn(ctx context.Context, conn net.Conn) (net.Conn, error) {
+	authContext, err := s.cfg.Authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	checker := authContext.Checker
+	certExpires := authContext.Identity.GetIdentity().Expires
+
+	var disconnectCertExpired time.Time
+	if !certExpires.IsZero() && checker.DisconnectExpiredCert() {
+		disconnectCertExpired = certExpires
+	}
+
+	idleTimeout := authContext.Checker.AdjustClientIdleTimeout(time.Duration(0))
```

Comment:
I think that instead of passing `time.Duration(0)` we should get the cluster level client timeout (from ClusterConfig) like it's done in other places.